### PR TITLE
Feature: Saga provide a switch to skip branch report on branch success, in order to optimize performance.

### PR DESCRIPTION
**Pull request by long187**
### Ⅰ. Describe what this PR did
Saga provide a switch to skip branch report on branch success, in order to optimize performance.

### Ⅱ. Does this pull request fix one issue?
#2150

### Ⅳ. Describe how to verify it
add config attribute `client.rm.report.success.enable=false`  file.conf
```
client {
  rm {
    report.success.enable = false
  }
```

execute unit test `StateMachineDBTests.java`

### CODE REVIEW

File: saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java [hunk 1]
```diff
@@ -48,6 +48,7 @@
     private String dbType;
     private int transOperationTimeout = DEFAULT_TRANS_OPER_TIMEOUT;
     private SagaTransactionalTemplate sagaTransactionalTemplate;
+    private boolean skipBranchReportOnSuccess;
```
Comment (slievrly) [inline saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java]: https://github.com/seata/seata/blob/a1a51e0c3d374ed0f9604012e9aac1f9822a7495/script/client/spring/application.yml#L9   There is a similar function switch in AT mode. If this configuration can be combined?

Comment (long187) [reply saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java]: Done

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java [hunk 2]
```diff
@@ -106,7 +106,7 @@
     /**
      * The constant CLIENT_REPORT_SUCCESS_ENABLE.
      */
-    public static final String CLIENT_REPORT_SUCCESS_ENABLE = CLIENT_PREFIX + "report.success.enable";
+    public static final String CLIENT_REPORT_SUCCESS_ENABLE = CLIENT_RM_PREFIX + "report.success.enable";
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: need to reset this file, `CLIENT_RM_PREFIX = CLIENT_PREFIX + "rm."`  The configuration name here is a dash, not a dot. 
image
image

Comment (long187) [reply core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: 8385F44E-C62E-40C2-8A37-EA5A97C2AEF3

**Pull request merged**